core/domain/collection_domain.py

```diff
@@ -907,132 +912,71 @@ def exploration_ids(self):
 
     @property
     def init_exploration_ids(self):
-        """Returns a list of exploration IDs that are starting points for this
-        collection (ie, they require no prior skills to complete). The order
-        of these IDs is given by the order each respective exploration was
-        added to the collection.
+        """Returns the first element in the node list of the collection, which
+           corresponds to the first node that the user would encounter.
 
         Returns:
-            list(str). List of exploration IDs.
+            list(str). A single element list of the exploration ID of the
+                first node.
         """
-        init_exp_ids = []
-        for node in self.nodes:
-            if not node.prerequisite_skill_ids:
-                init_exp_ids.append(node.exploration_id)
-        return init_exp_ids
-
-    def get_next_exploration_ids(self, completed_exploration_ids):
-        """Returns a list of exploration IDs for which the prerequisite skills
-        are satisfied. These are the next explorations to complete for a user.
-        If the list returned is empty and the collection is valid, then all
-        skills have been acquired and the collection is completed. If the input
-        list is empty, then only explorations with no prerequisite skills are
-        returned. The order of the exploration IDs is given by the order in
-        which each exploration was added to the collection.
+        if len(self.nodes) > 0:
+            return [self.nodes[0].exploration_id]
+        else:
+            return []
+
+    def get_next_exploration_id(self, completed_exp_ids):
```

Comment:
Either keep as get_next_exploration_ids or return a string. The former is more consistent with nonlinear stories.

Reply:
Yeah, I was back and forth on whether to return a string or list for init_exploration_id and get_next_exploration_id. If a string is returned for init_exploration_id, then they have to be converted to a list anyway when getting the nodes in playable order (as the next exploration ids are appended to it), but it would be better if both these functions (init_exploration_id and get_next_exploration_id) have the same return type, as some tests compare their return values directly, so I just made them both a single element list.
Shall I just change the names back to plural for these functions, or should both return a string?

Reply:
Done, changed to string, with singular 'id' in titles.